### PR TITLE
feat #1064: prompt to overwrite when rz upload hits a remote filename conflict

### LINKED ISSUE #1064: Bug: 使用rz上传文件，如果远端当前目录已有相同文件名，无法上传成功
## Steps to reproduce
1.

## Environment
- App: netcatty 1.1.9
- Platform: win32
- UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) netcatty/1.1.9 Chrome/144.0.7559.236 Electron/40.8.5 Safari/537.36

### PR DESCRIPTION
**Pull request by binaricat**
## Problem

Closes #1064. Uploading via `rz` silently fails when the remote directory already contains a file with the same name: lrzsz `rz` sends ZSKIP for existing files, and `handleUpload` treated the falsy `send_offer` result as "skipped" and `continue`d — so nothing uploaded and no feedback.

## Approach

Detect the conflict through a **separate SSH exec channel** (never touching the ZMODEM data stream) and prompt the user to overwrite, skip, or cancel, with an "apply to remaining conflicts" option. Overwrite is implemented by `rm`-ing the remote file via that same exec channel and then sending the ZMODEM offer normally — so no changes to the ZMODEM protocol / zmodem.js.

- **SSH only.** The detection/removal callbacks are supplied only by the SSH bridge; local/telnet/serial fall back to today's behavior.
- **Fail-safe.** If rz's cwd can't be resolved, the remote isn't Linux (no `/proc`), or the exec lacks permission (e.g. after `su`/`sudo` into another user) → silently fall back to today's behavior (rz skips). No regression.

## How it works

1. After the file picker, `handleUpload` calls `probeReceiveConflicts(basenames)` — an exec script that finds the running `rz` process's cwd (reusing the #1065 process-tree walk to find the foreground process, then `/proc/<pid>/cwd`) and `[ -e ]`-tests each basename there.
2. For each conflict, `requestOverwriteDecision(filename)` round-trips to the renderer (main→renderer→main IPC) and shows a dialog.
3. A pure, unit-tested `buildUploadPlan(names, existing, resolve)` turns the decisions into `{ filesToOffer, filesToRemove, aborted }` (handles overwrite/skip/cancel + applyToRest).
4. `removeRemoteFiles(paths)` `rm -f`s the overwrite targets; the offer loop then runs over the filtered list.

All remote paths derive from local basenames + the resolved cwd and are shell-quoted (`sh -c '…' sh "$@"`), so a crafted filename can't inject shell code; removal targets are never taken from the server's echoed output.

## Testing

- **Unit:** `buildUploadPlan` — 6 tests (overwrite/skip/cancel/applyToRest/no-conflict/selectivity). Full suite green: **1143 pass / 0 fail**.
- **Live server:** the exec probe + remove verified against a real `rz` waiting in `/tmp` with a duplicate present → probe returns `{dir:/tmp, existing:[dup]}` (ignores a non-existent name); the `rm` deletes it.
- **Build/lint/types:** `npm run build` passes; ESLint clean; `tsc` adds zero new errors.
- An independent full-feature code review (security/lifecycle/edge cases) was incorporated — including a fix for a test-mock regression (`ipcMain.on`) and i18n of the dialog across en/zh-CN/ru.

## Known limitations (documented, out of scope)

- Cross-uid `su`/`sudo` into another user's dir: the exec (login user) can't read/`rm` there → falls back to today's behavior.
- Non-Linux remotes (no `/proc`) → fallback.
- If the session drops while the dialog is open, the pending decision self-resolves to "skip" after 120s.

> Note: the end-to-end GUI click-through (real upload + dialog) should be confirmed in the running app before merge; every component + the exec mechanism are verified above.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: electron/bridges/zmodemHelper.cjs [hunk 1]
```diff
@@ -524,29 +550,63 @@ async function handleUpload(zsession, opts) {
...
+    } catch (err) {
+      if (err instanceof Error && err.message === "Transfer cancelled") throw err;
+      console.warn("[ZMODEM] conflict probe failed; proceeding:", err?.message || err);
+    }
+  }
+
+  const offerSet = new Set(plan.filesToOffer);
+  const offers = filePaths
+    .map((filePath, i) => ({ filePath, stat: fileStats[i], name: allNames[i] }))
+    .filter((o) => offerSet.has(o.name));
```
[the remark this answers is not in the document]
Comment (binaricat) [reply electron/bridges/zmodemHelper.cjs]: Fixed in the latest commit. The upload plan is now **index-based** instead of keyed by basename, so each selected file's decision is preserved by position.

`buildUploadPlan` returns `{ offerIndices, removeIndices, aborted }` (indices into the original `names`), and `handleUpload` maps them back directly:
```js
const offers = plan.offerIndices.map((i) => ({ filePath: filePaths[i], stat: fileStats[i], name: allNames[i] }));
```
No `Set`-by-name collapse anywhere. rm targets are de-duplicated separately (`[...new Set(...)]`) so an overwrite chosen for two same-named entries only issues one `rm`.

Added a regression test: two entries sharing a basename where the first is skipped and the second overwritten → `offerIndices: [1], removeIndices: [1]` (the skipped file is not sent).

Note: in practice `dialog.showOpenDialog` multi-select returns files from a single directory, so `allNames` can't currently contain duplicate basenames — this wasn't reachable today. But the index-based mapping removes the latent fragility and is correct if the upload source ever changes. Full suite: 1144 pass / 0 fail.

**Comment by binaricat** [comment]
@codex review

File: electron/bridges/zmodemHelper.cjs [hunk 2]
```diff
@@ -524,29 +553,60 @@ async function handleUpload(zsession, opts) {
...
+      if (probe && probe.dir && Array.isArray(probe.existing) && probe.existing.length > 0) {
+        plan = await buildUploadPlan(allNames, probe.existing, opts.requestOverwriteDecision);
+        if (plan.aborted) {
+          try { zsession.abort(); } catch { /* ignore */ }
+          abortRemoteProcess(opts.writeToRemote);
+          throw new Error("Transfer cancelled");
+        }
+        if (plan.removeIndices.length && opts.removeRemoteFiles) {
+          const base = probe.dir.replace(/\/+$/, "");
+          const targets = [...new Set(plan.removeIndices.map((i) => `${base}/${allNames[i]}`))];
```
[the remark this answers is not in the document]
Comment (binaricat) [reply electron/bridges/zmodemHelper.cjs:573]: Pushing back on this one after checking it carefully — I don't think a change is warranted:

1. **Not reachable.** `dialog.showOpenDialog` multi-select returns files from a single directory, so `allNames` cannot contain duplicate basenames. With distinct basenames the paths are distinct, so the `[...new Set(...)]` dedup is a no-op in every reachable case.

2. **Dropping the dedup wouldn't fix the described behavior anyway.** All removals run *before* the offer loop (`removeRemoteFiles` at line ~575, the `for` loop at ~589). So even with two `rm`s of the same path, the first offer recreates `<dir>/x.txt` and `rz` skips the second offer — identical outcome. The dedup isn't the cause.

3. **Inherent collision.** Two local files map to one remote path (`rz` writes by basename to its cwd). The remote can only hold one — "last writer wins" is the only coherent result; you can't keep both. Making the *last* overwrite win would require `rm`-before-each-send interleaving, which is real added complexity for an input that can't occur today (YAGNI).

If a future change ever introduces a multi-directory upload source, the correct handling would be "collapse duplicate basenames keeping the last occurrence" (so we don't even send files destined to be overwritten) — happy to add that then. For this PR I'm leaving the dedup as a harmless defensive no-op. The per-file *decision* integrity that the P1 was about is preserved by the index-based plan.
